protocols/kad/src/behaviour.rs

```diff
@@ -546,7 +546,7 @@ where
     pub fn kbucket<K>(&mut self, key: K)
         -> Option<kbucket::KBucketRef<'_, kbucket::Key<PeerId>, Addresses>>
     where
-        K: Borrow<[u8]> + Clone
```

Comment:
The removal of `Borrow` constraints is very unfortunate because it weakens the semantics, i.e. `Into<Vec<u8>>` is too lax of a constraint. The generic code in `libp2p-kad` (and maybe elsewhere in libp2p where `Borrow` is used) wants a byte representation of some key type `K` whose `Eq`, `Ord` and `Hash` are consistent with those same impls for `K` in order for there to be no surprises (which we had in the past). That requirement is lost with `Into<Vec<u8>>` and in the absence of `Borrow` may probably only be recovered by a custom trait or excessive, repeated documentation that may more easily be overlooked compared to the more well-known semantics of `std::borrow::Borrow`.

Reply:
Thanks for the feedback, I've changed my approach with the newest commit. Now the `Borrow` is preserved.